File: src/main/java/sh/twenty/octane/commands/HealCommand.java
package sh.twenty.octane.commands;

import org.bukkit.entity.Player;
import sh.twenty.octane.Main;
import sh.twenty.octane.utils.ChatUtil;

public class HealCommand extends PlayerCommand {

    public HealCommand(Main plugin) {
        super(plugin, "octane.heal");
    }

    @Override
    public boolean execute(Player player, String[] args) {

        if (args.length >= 1){


            if(!player.hasPermission("octane.heal.others")){
                String rawMessage = plugin.getConfigManager().getMessages().getString("no-permission", "&cNo permission!");
                player.sendMessage(ChatUtil.color(rawMessage));
                return false;
            }

            Player target = plugin.getServer().getPlayer(args[0]);

            if (target == null){
                sendPrefixedMessage(player, "&cCould not find player: " + args[0]);
                return false;
            }

            target.setHealth(20.0);

            sendPrefixedMessage(player, "&7" + target.getName() + " has been healed.");
            sendPrefixedMessage(target, "&7You were healed by " + player.getName() + ".");
            return true;
        }
        player.setHealth(20.0);
        sendPrefixedMessage(player, "&7You have been fully healed.");
        return true;
    }
}

File: src/main/kotlin/sh/twenty/octane/commands/HealCommand.kt
package sh.twenty.octane.commands

import dev.jorel.commandapi.kotlindsl.commandAPICommand
import dev.jorel.commandapi.kotlindsl.playerArgument
import dev.jorel.commandapi.kotlindsl.playerExecutor
import org.bukkit.attribute.Attribute
import org.bukkit.entity.Player
import sh.twenty.octane.OctanePlugin

class HealCommand(private val plugin: OctanePlugin) {
    fun register() = commandAPICommand("heal") {
        withPermission("octane.heal")
        playerArgument("target", optional = true)
        playerExecutor { player, args ->
            val msgs = plugin.configManager.messages
            val target = args["target"] as? Player

            if (target != null && !player.hasPermission("octane.heal.others")) {
                player.sendRichMessage(msgs.prefix + msgs.noPermission)
                return@playerExecutor
            }

            val finalTarget = target ?: player
            finalTarget.health = finalTarget.getAttribute(Attribute.GENERIC_MAX_HEALTH)?.value ?: 20.0

            if (target != null) {
                player.sendRichMessage(msgs.prefix + msgs.healOther.replace("%player%", finalTarget.name))
                finalTarget.sendRichMessage(msgs.prefix + msgs.healTarget.replace("%sender%", player.name))
            } else {
                player.sendRichMessage(msgs.prefix + msgs.healSelf)
            }
        }
    }
}
